docs/error-handling.md

```diff
@@ -0,0 +1,242 @@
+
+#### Bad Example (`clientLoader` without `handleLoaderError`)
+
+```typescript
+// This is not recommended
+export async function clientLoader({ params }: ClientLoaderFunctionArgs) {
+  const { dapper } = getLoaderTools();
+  try {
+    const team = await dapper.getTeam(params.teamId);
+    // Awaiting all follow-up data forces the UI to block on the clientLoader.
+    const members = await dapper.getTeamMembers(params.teamId);
+    return { team, members };
+  } catch (error) {
+    // Errors will bubble to Remix but we miss the user-facing payload mapping.
+    throw error;
+  }
+}
+```
+
+#### Good Example (`clientLoader` with `handleLoaderError`)
+
+```typescript
+// This is the recommended approach
+export async function clientLoader({ params }: ClientLoaderFunctionArgs) {
+  const { dapper } = getLoaderTools();
+
+  let team;
+  try {
+    // Await the first fetch only because the follow-up request depends on it.
+    team = await dapper.getTeam(params.teamId);
+  } catch (error) {
+    // This provides a consistent, user-facing error.
+    handleLoaderError(error);
+  }
+
+  // Kick off dependent work without awaiting so Suspense can resolve it downstream.
+  const members = dapper.getTeamMembers(team.slug);
+
+  return {
+    team,
+    members,
+  };
+}
+```
+
+---
+
+## `UserFacingError`
+
+When an error is thrown, `handleLoaderError` uses `throwUserFacingPayloadResponse` to create a `Response` with a JSON payload containing a `UserFacingError`. This structured error object ensures that the UI can present a helpful message.
+
+The `UserFacingError` interface is defined in the `thunderstore-api` package.
+
+```typescript
+// packages/thunderstore-api/src/errors/userFacingError.ts
+export interface UserFacingError {
+  headline: string;
+  description?: string;
+  category?: UserFacingErrorCategory;
+  status?: number;
+  context?: Record<string, unknown>;
+}
+```
+
+-   **Definition**: [`packages/thunderstore-api/src/errors/userFacingError.ts`](../../packages/thunderstore-api/src/errors/userFacingError.ts)
+-   **Response Helper**: [`apps/cyberstorm-remix/cyberstorm/utils/errors/userFacingErrorResponse.ts`](../../apps/cyberstorm-remix/cyberstorm/utils/errors/userFacingErrorResponse.ts)
+
+---
+
+## React Router Related Error Handling
+
+### Usage in ErrorBoundary
+
+A route's `ErrorBoundary` can use `resolveRouteErrorPayload` to parse the error from the `Response` and render a user-friendly message.
+
+```typescript
+// apps/cyberstorm-remix/app/some-route.tsx
+import { useRouteError } from "react-router";
+import { resolveRouteErrorPayload } from "cyberstorm/utils/errors/resolveRouteErrorPayload";
+import { NimbusErrorBoundaryFallback } from "cyberstorm/utils/errors/NimbusErrorBoundary";
+
+export function ErrorBoundary() {
+  const error = useRouteError();
+  const payload = resolveRouteErrorPayload(error);
+
+  return (
+    <NimbusErrorBoundaryFallback
+      error={error}
+      title={payload.headline}
```

Comment:
Could payload.headline be renamed to payload.title so it's in line with usage (and also shorter).